diff --git a/OpenTabletDriver.Configurations/Configurations/XP-Pen/Deco Pro Medium.json b/OpenTabletDriver.Configurations/Configurations/XP-Pen/Deco Pro Medium.json
index aa7e9318..7b76433d 100644
--- a/OpenTabletDriver.Configurations/Configurations/XP-Pen/Deco Pro Medium.json	
+++ b/OpenTabletDriver.Configurations/Configurations/XP-Pen/Deco Pro Medium.json	
@@ -13,7 +13,17 @@
     },
     "AuxiliaryButtons": {
       "ButtonCount": 8
-    }
+    },
+    "Wheels": [
+      {
+        "RelativeWheelSteps": 24,
+        "ButtonCount": 0
+      },
+      {
+        "RelativeWheelSteps": 24,
+        "ButtonCount": 0
+      }
+    ]
   },
   "DigitizerIdentifiers": [
     {
@@ -21,7 +31,7 @@
       "ProductID": 2308,
       "InputReportLength": 10,
       "OutputReportLength": 10,
-      "ReportParser": "OpenTabletDriver.Configurations.Parsers.XP_Pen.XP_PenReportParser",
+      "ReportParser": "OpenTabletDriver.Configurations.Parsers.XP_Pen.XP_PenReportParserV2",
       "OutputInitReport": [
         "ArAE"
       ]
@@ -30,4 +40,4 @@
   "Attributes": {
     "libinputoverride": "1"
   }
-}
+}
\ No newline at end of file
diff --git a/OpenTabletDriver.Configurations/Parsers/XP_Pen/XP_PenAuxReport.cs b/OpenTabletDriver.Configurations/Parsers/XP_Pen/XP_PenAuxReport.cs
index 4d046ddf..4a7286a8 100644
--- a/OpenTabletDriver.Configurations/Parsers/XP_Pen/XP_PenAuxReport.cs
+++ b/OpenTabletDriver.Configurations/Parsers/XP_Pen/XP_PenAuxReport.cs
@@ -5,7 +5,7 @@ namespace OpenTabletDriver.Configurations.Parsers.XP_Pen
 {
     public struct XP_PenAuxReport : IAuxReport, IRelativeWheelReport
     {
-        public XP_PenAuxReport(byte[] report, int auxIndex = 2, int wheelIndex = 7)
+        public XP_PenAuxReport(byte[] report, int auxIndex = 2, int wheelIndex = 7, int secondWheelBitOffset = 0)
         {
             Raw = report;
 
@@ -35,10 +35,11 @@ namespace OpenTabletDriver.Configurations.Parsers.XP_Pen
 
             // 0x01 for 1st wheel clockwise, 0x02 for counterclockwise, verified on XP Pen Artist 13.3 Pro V2
             // 0x10 for 2nd wheel clockwise, 0x20 for counterclockwise, verified on XP Pen Artist 22R Pro
+            // 0x04 for 2nd wheel clockwise, 0x08 for counterclockwise, verified on XP Pen Deco Pro Medium
             AnalogDeltas =
             [
                 report[wheelIndex].IsBitSet(0) ? 1 : report[wheelIndex].IsBitSet(1) ? -1 : 0,
-                report[wheelIndex].IsBitSet(4) ? 1 : report[wheelIndex].IsBitSet(5) ? -1 : 0,
+                report[wheelIndex].IsBitSet(4 + secondWheelBitOffset) ? 1 : report[wheelIndex].IsBitSet(5 + secondWheelBitOffset) ? -1 : 0,
             ];
         }
 
diff --git a/OpenTabletDriver.Configurations/Parsers/XP_Pen/XP_PenReportParserV2.cs b/OpenTabletDriver.Configurations/Parsers/XP_Pen/XP_PenReportParserV2.cs
new file mode 100644
index 00000000..57adb433
--- /dev/null
+++ b/OpenTabletDriver.Configurations/Parsers/XP_Pen/XP_PenReportParserV2.cs
@@ -0,0 +1,25 @@
+using System.Diagnostics.CodeAnalysis;
+using OpenTabletDriver.Plugin.Tablet;
+
+namespace OpenTabletDriver.Configurations.Parsers.XP_Pen
+{
+    [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
+    public class XP_PenReportParserV2 : IReportParser<IDeviceReport>
+    {
+        public IDeviceReport Parse(byte[] report)
+        {
+            if (report[1] == 0xC0)
+                return new OutOfRangeReport(report);
+
+            if (report[1].IsBitSet(4))
+                return new XP_PenAuxReport(report, secondWheelBitOffset: -2);
+
+            if (report.Length >= 12)
+                return new XP_PenTabletOverflowReport(report);
+            else if (report.Length >= 10)
+                return new XP_PenTabletReport(report);
+            else
+                return new TabletReport(report);
+        }
+    }
+}
